core/FrontController.php

```diff
@@ -493,10 +493,14 @@ protected function prepareDispatch($module, $action, $parameters)
 
     protected function handleMaintenanceMode()
     {
-        if ((Config::getInstance()->General['maintenance_mode'] != 1) || Common::isPhpCliMode()) {
+        if ((Config::getInstance()->General['maintenance_mode'] != 1) || Common::isPhpCliMode() ) {
             return;
         }
-        Common::sendResponseCode(503);
+
+        // as request matomo behind load balancer should not return 503. https://github.com/matomo-org/matomo/issues/18054
+        if (Config::getInstance()->General['multi_server_environment'] != 1) {
```

Comment:
I was wondering if we maybe should replace the usage of Config with the `GeneralConfig::getConfigValue`, when adjusting such code.
It's not really needed, but might be good at some point to only use `GeneralConfig`